File: rotate-array/rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int n =nums.size();
        k = k%n;
        int d=n-k;
        vector<int> vec;
        for(int i=0; i<d; i++)
        {
            vec.push_back(nums[i]);
        }
        for(int i=d; i<n; i++)
        {
            nums[i-d]=nums[i];
        }
        int j=0;
        for(int i=k; i<n; i++)
        {
            nums[i]=vec[j];
            j++;
        }
    }
};

File: 189-rotate-array/rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        int[] temp = new int[k];
        int n = nums.length;
        if(n < k) k = k % n; 

    // Step1 - Store k elements in temp array
        for(int i=n-k; i<=n-1; i++){
            temp[i-(n-k)] = nums[i];
        }   

    // Step2 - Right Shift by k position
        for(int i=n-k-1; i>=0; i--){
            nums[i+k]= nums[i];
        }

    // Step3 - Copy the K elements to start
        for(int i=0; i<k; i++){
            nums[i] = temp[i];
        }

    }
}